modules/auxiliary/gather/zoomeye_domain.rb

```diff
@@ -0,0 +1,130 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+#
+##
+
+class MetasploitModule < Msf::Auxiliary
+    include Msf::Auxiliary::Report
+
+    def initialize(info={})
+        super(update_info(
+                info,
+                "name" => "ZoomEye Domain",
+                "Description" => %q{
+            The module use the ZoomEye API to search ZoomEye.ZoomEye is a cyberspace search engine,
+            users can search for network devices using a browser.
+            },
+                'Author' => ['wh0am1i'],
+                "References" => [
+                  ["URL", "https://www.zoomeye.org/api/doc"],
+                  ["URL", "https://github.com/knownsec/ZoomEye-python"]
+                ],
+                "License" => MSF_LICENSE
+              ))
+
+        register_options(
+          [
+            OptString.new('APIKEY', [true, 'The ZoomEye API KEY']),
+            OptString.new('QUERY', [true, 'The ZoomEye dork']),
+            OptInt.new('PAGE', [true, "Max amount of pages to collect", 1]),
+            OptInt.new("SOURCE", [true, "Domain search type", 0]),
+            OptBool.new('OUTFILE', [false, 'A filename to store ZoomEye search raw data']),
+            OptBool.new('DATABSE', [false, 'Add search results to the database'])
+          ]
+        )
+    end
+
+    def zoomeye_resolvable?
+        begin
+            Rex::Socket.resolv_to_dotted("api.zoomeye.org")
+        rescue RuntimeError, SocketError
+            return false
+        end
+        true
+    end
+
+    def parse_domain_info(data)
+        tab = Rex::Text::Table.new(
+          "Header" => "Web Search Result",
+          "Indent" => 1,
+          "Columns" => ['IP', 'NAME', 'TIMESTAMP']
+        )
+        data.each do |item|
+            name = item['name']
+            timestamp = item['timestamp']
+            ip = item['ip']
+
+            tab << [ip, name, timestamp]
+
+            report_host(:host => ip,
+                        :name => name,
+                        :comments => 'Added from ZoomEye Domain Search',
+                        ) if datastore['DATABASE']
+        end
+        print_line("#{tab}")
+    end
+
+    def save_raw_data(query, data, page)
+        filename = query.gsub(/[^a-zA-Z ]/,'_')
+        ::File.open("#{filename}_#{page}.json", "wb") do |f|
+            f.write(ActiveSupport::JSON.encode(data))
+        end
+    end
+
+    def domain_search(apikey, query, page, s_type)
+        request = Rex::Proto::Http::Client.new('api.zoomeye.org', 443, {}, true)
+        request.connect
+
+        begin
+            response = request.request_cgi({
```

Comment:
Following on the advice from https://github.com/rapid7/metasploit-framework/pull/15429/files#r670535449 to mimic the [shodan functionality](https://github.com/rapid7/metasploit-framework/blob/0ca436d1e80acb0eb251e24f28e6b261458380c4/modules/auxiliary/gather/shodan_search.rb#L52-L63) I believe this line would be:

```suggestion
            response = send_reqest_cgi({
```

Reply:
Thanks for all your suggestions, I will modify the code according to the suggestions later, and then resubmit the code, thank you very much!

Reply:
`send_reqest_cgi` is my favorite.